.github/workflows/fetch-dev-artifacts.yaml

```diff
@@ -0,0 +1,97 @@
+    
+    - name: Determine latest neo4j CI version
+      run: |
+          neo4j_version_base=$(cat neoBaseVersion.txt)
+          echo "neo4j_version_base=$neo4j_version_base" >> $GITHUB_ENV
+          echo "Found neo4j_version_base=$neo4j_version_base"
+
+    - name: fetch parser artifacts from latest neo4j CI
+      run: |
+          aws --no-cli-pager codeartifact get-package-version-asset --domain build-service-live --repository ci-live --format maven --namespace org.neo4j --package cypher-v25-antlr-parser --package-version $neo4j_version_base-SNAPSHOT --asset $(aws --no-cli-pager codeartifact list-package-version-assets --domain build-service-live --repository ci-live --format maven --namespace org.neo4j --package cypher-v25-antlr-parser --package-version $neo4j_version_base-SNAPSHOT --query 'assets[?ends_with(name, `grammar.zip`)].name | [0]' --output text --domain-owner ${{ secrets.AWS_ACCOUNT_ID }}) grammar.zip --domain-owner ${{ secrets.AWS_ACCOUNT_ID }}
+          
+    - name: fetch semantic analysis artifacts from latest neo4j CI
+      run: | 
+          aws --no-cli-pager codeartifact get-package-version-asset --domain build-service-live --repository ci-live --format maven --namespace com.neo4j --package semantic-analysis-js --package-version $neo4j_version_base-SNAPSHOT --asset $(aws --no-cli-pager codeartifact list-package-version-assets --domain build-service-live --repository ci-live --format maven --namespace com.neo4j --package semantic-analysis-js --package-version $neo4j_version_base-SNAPSHOT --query 'assets[?ends_with(name, `artifacts.zip`)].name | [0]' --output text --domain-owner ${{ secrets.AWS_ACCOUNT_ID }}) semantic-analysis-js.zip --domain-owner ${{ secrets.AWS_ACCOUNT_ID }}
+    
+    - name: unpack grammar artifacts
+      run: |
+           unzip -j -o grammar.zip \
+            "$(unzip -l grammar.zip | awk '{print $4}' | grep "Cypher25Lexer.g4$")" \
+            "$(unzip -l grammar.zip | awk '{print $4}' | grep "Cypher25Parser.g4$")" \
+            -d ./packages/language-support/src/antlr-grammar
+
+    - name: unpack semantic analysis
+      run: |
+        unzip -j -o "semantic-analysis-js.zip" \
+         "$(unzip -l semantic-analysis-js.zip | awk '{print $4}' | grep "classes.js$")" \
+         -d ./packages/language-support/src/syntaxValidation
+
+    
+    - name: rename semantic analysis
+      run: mv -f "./packages/language-support/src/syntaxValidation/classes.js" "./packages/language-support/src/syntaxValidation/semanticAnalysis.js"
+
+              
+    - name: clean up zips
+      run: rm -f grammar.zip && rm -f semantic-analysis-js.zip
+
+    - name: Create New Branch Name
+      run: |
+        BRANCH_NAME="update-artifacts-$(date +%Y%m%d-%H%M%S)"
+        echo "BRANCH_NAME=$BRANCH_NAME" >> $GITHUB_ENV
+
+    - name: Commit and Push Changes
+      run: |
+        git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
```

Comment:
I don't think this is needed. Source: https://stackoverflow.com/questions/57921401/push-to-origin-from-github-action/58393457#58393457